### ISSUE TITLE
stderr shows up in nrepl-messages, with no id

### ISSUE THREAD

**Issue by tommy-mor** [clojure-emacs/cider#1598]
I am using clojure quil to make drawings, but runtime errors from quil are being sent to nrepl-messages.
For example of runtime errors from quil in nrepl-messages:
http://pastebin.com/hu1eN5RM
(I think these are stdout because they have \n's in them, and when using different environments, they show up in stdout)
To me (although I am no expert) it seems like this is the correct behavior, as normal clojure errors show up in nrepl-messages aswell.
The problem is that clojure **quil errors do not have an id attribute** , so when they show up in the nrepl-messages buffer, I get
`error in process filter: nREPL: No response handler with id nil found`
This is super annoying, because everytime that message shows up, my editor lags for a few hundered milliseconds, which is incredibly annoying because when multiple errors show up it can lag for ~3 seconds.
This happens when running any type of quil project.
I do not know if the problem is because of CIDER's handling of stderr. It also might be leinigen with it's handling of stderr. I do not know. (sorry if I put this bug report on the wrong project)
## Environment & Version information

`CIDER 0.11.0snapshot`
using cider-nrepl 0.11.0snapshot aswell
using nREPL 0.2.10
this is when using a standalone repl

I also get it using the cider-jack-in command, using

```
;; CIDER 0.11.0snapshot (package: 20160229.922), nREPL 0.2.12
;; Clojure 1.7.0, Java 1.7.0_79
```

I just want the editor lags to go away, which means redirecting or stopping these idless repl messages

**Comment by bbatsov** [comment]
Is this still an issue? Usually such problems are caused by some version mismatch.

**Comment by Malabarba** [comment]
If it's really the case that nrepl can send messages without an ID, then we definitely shouldn't complain about it in cider. But I think it's strange that nrepl would do that.

**Comment by bbatsov** [comment]
I'm pretty sure something like this should never happen, but this might be some nREPL bug.

**Comment by Tommy-tf2** [comment]
Still is an issue, but I will change around versions of packages. It also may be worth noting that it is stderr.
Any ideas on how to fix this within nrepl? Is there any config settings I could change?

**Comment by Tommy-tf2** [comment]
I also have already tried difserent packages,  I changed from nrepl 2.12 and 2.10

**Comment by bbatsov** [comment]
Btw, can you post here the messages you see in `*nrepl-messages ...*`?

**Comment by Tommy-tf2** [comment]
http://pastebin.com/hu1eN5RM
I can get more later. I think they are runtime errors of processing/quil, the library I'm using

**Comment by bbatsov** [comment]
@Malabarba I wonder if this is some bug in our out middleware.

**Comment by Malabarba** [comment]
Anything's possible, but nothing jumps out to me by looking at the code.

**Comment by bbatsov** [comment]
It seems to me the responses are not associated with the init request for some reason (assuming our middleware is causing this, that is). Obviously precise reproduction steps would help.

**Comment by Tommy-tf2** [comment]
Personnally, I think it is a mistake/error in the way I set it up. I will try to reproduce the bug on a different computer

**Comment by jackrusher** [comment]
Firstly, I first want to think you for all the good work you lot do on Cider. I'm seeing the same thing, and it has made quil unusual for me. *Any* error in the `draw` loop requires at minimum a `cider-quit` + `cider-connect` (with liberal doses of `C-g` to get there), at most a complete restart of emacs. None of this happened prior to upgrading to cider-11.0.

**Comment by Malabarba** [comment]
I just managed to get this too when working with ClojureScript. `println`s that happen in the browser are sent to the cljs connection but don't contain an id:

    (<-- 
      out  "Content of the print"
      session  "49a3f8eb-d5e6-4844-aac0-2589c4dbc78f"
    )

Maybe we should just accept that this will happen for `out` and `err`? Instead of always complaining, we could handle these id-less messages gracefully, and only complain if the contain some other key besides `session`, `out`, and `err`.

**Comment by bbatsov** [comment]
I'm fine with adding some workaround for now, but we should also identify what's causing this. I cannot imagine how some response didn't originate as the result of a request.

**Comment by bbatsov** [comment]
Also I'd font-lock such requests differently in *nrepl-messages* so they stand out a bit.

**Comment by triss** [comment]
Just thought I'd drop a not here to say this is a also problem for me at the moment. I'm using latest version of Quil and CIDER integrated with spacemacs.

Did anyone manage to pull together a quick fix?

**Comment by bbatsov** [comment]
> Did anyone manage to pull together a quick fix?

Nope.

**Comment by Malabarba** [comment]
You can go into nrepl-client.el and comment out the code that signals this warningl=. :-P

**Comment by bhagany** [comment]
This is fixed in #387
